Sources/SpelledPitch/PitchSpeller/Wetherfield/SpellingInverter.swift

```diff
@@ -0,0 +1,430 @@
+//
+//  SpellingInverter.swift
+//  SpelledPitch
+//
+//  Created by Benjamin Wetherfield on 2/14/19.
+//
+
+import DataStructures
+import Pitch
+
+struct SpellingInverter {
+    
+    // MARK: - Instance Properties
+    
+    // The pre-assigned `DirectedGraph` which stands in for the `FlowNetwork` that
+    // would solve to the set of spellings passed into the `SpellingInverter`.
+    var flowNetwork: DirectedGraph<PitchSpeller.AssignedNode>
+    
+    let pitchClass: (Int) -> Pitch.Class?
+}
+
+extension SpellingInverter {
+    
+    // MARK: - Type Aliases
+    
+    typealias AssignedEdge = OrderedPair<PitchSpeller.AssignedNode>
+    typealias UnassignedEdge = OrderedPair<PitchSpeller.UnassignedNode>
+    typealias PitchedEdge = UnorderedPair<FlowNode<Cross<Pitch.Class,Tendency>>>
+}
+
+extension SpellingInverter {
+    
+    // MARK: - Initializers
+    
+    init(spellings: [Int: Pitch.Spelling], parsimonyPivot: Pitch.Spelling = .d) {
+        self.flowNetwork = DirectedGraph(internalNodes: internalNodes(spellings: spellings))
+        self.pitchClass = { int in spellings[int]?.pitchClass }
+        
+        let specificEdgeScheme: DirectedGraphScheme<PitchSpeller.UnassignedNode> = (upDownEdgeScheme.pullback(nodeMapper) + sameEdgeScheme.pullback(nodeMapper))
+            * connectDifferentInts
+        
+        let sameIntEdgesScheme: DirectedGraphScheme<PitchSpeller.UnassignedNode> =
+            sameIntsScheme * connectSameInts
+        
+        let specificSourceScheme: DirectedGraphScheme<PitchSpeller.UnassignedNode> =
+            sourceEdgeLookupScheme.pullback(nodeMapper)
+        
+        let specificSinkScheme: DirectedGraphScheme<PitchSpeller.UnassignedNode> =
+            sinkEdgeLookupScheme.pullback(nodeMapper)
+        
+        let allSchemes: [DirectedGraphScheme<PitchSpeller.UnassignedNode>] = [
+            specificEdgeScheme,
+            sameIntEdgesScheme,
+            specificSourceScheme,
+            specificSinkScheme
+        ]
+        
+        let maskScheme: DirectedGraphScheme<PitchSpeller.AssignedNode> =
+            allSchemes.reduce(DirectedGraphScheme { _ in false }, +)
+                .pullback({ $0.unassigned })
+        
+        // Apply masking of specific manifestations of the general global adjacency schemes
+        self.flowNetwork.mask(maskScheme)
+    }
+}
+
+extension SpellingInverter {
+    
+    // MARK: - Computed Properties
+    
+    /// - Returns: A concrete distribution of weights to satisfy the weight relationships delimited by
+    /// `weightDependencies` or `nil` if no such distribution is possible, i.e. there are cyclical
+    /// dependencies between edge types. In the latter case, the spellings fed in are *inconsistent*.
+    /// Weights are parametrized by `Pitch.Class` and `Tendency` values.
+    var weights: [PitchedEdge: Double] {
+        
+        let dependencies = pitchedDependencies
```

Comment:
Is this necessary? I tried substituting `dependencies` with `pitchedDependencies` and all the tests passed.

In this case, `pitchedDependencies` is a Copy-On-Write `Dictionary`, so the copy is made for you, only when needed.

Reply:
Does a computed property get recomputed every time it is called? I think that's why I stashed it

Reply:
Oooooh boy, good point. I must have overlooked that it has a non-trivial implementation. Not sure what I was thinking.

An idiom which I haven't done a good job of following myself (though perhaps I aspire to), is to use functions for any properties which are not O(*1*), so maybe we should use:

```Swift
func pitchedDependencies() -> [Pitch: Double]
```

Reply:
Sorry about that 🤦‍♂

Reply:
I like that convention!